File: src/lib/app-mode.ts
import { derived } from 'svelte/store';
import { config } from '$lib/stores';

export type OpenWebUIAppMode = 'default' | 'pageindex';

export const normalizeAppMode = (value?: string | null): OpenWebUIAppMode => {
	return value === 'pageindex' ? 'pageindex' : 'default';
};

export const appMode = derived(config, ($config) => normalizeAppMode($config?.app_mode));

export const is_default_mode = derived(appMode, ($appMode) => $appMode === 'default');
export const is_pageindex_mode = derived(appMode, ($appMode) => $appMode === 'pageindex');

File: backend/open_webui/utils/app_mode.py
from typing import Optional

from fastapi import Request

from open_webui.config import OPENWEBUI_APP_MODE


def get_app_mode(request: Optional[Request] = None) -> str:
    if request is not None:
        app_mode = getattr(getattr(request, "app", None), "state", None)
        if app_mode is not None:
            value = getattr(request.app.state, "APP_MODE", None)
            if isinstance(value, str) and value:
                return value
    return OPENWEBUI_APP_MODE


def is_default_mode(request: Optional[Request] = None) -> bool:
    return get_app_mode(request) == "default"


def is_pageindex_mode(request: Optional[Request] = None) -> bool:
    return get_app_mode(request) == "pageindex"
